### ISSUE TITLE
Bug: Ubuntu 22.04 fails to run comtraya 0.9.0 citing unable to find GLIBC_2.38 or GLIBC_2.39

### ISSUE THREAD

**Issue by airtonix** [comtrya/comtrya#493]
**Description**

```
❯ comtrya apply -m shells/zsh.yml

A new version of comtrya is available: v0.8.7 -> v0.9.0
Changelog: https://github.com/comtrya/comtrya/releases/tag/v0.9.0
Run to update: curl -fsSL https://get.comtrya.dev | sh

.dotfiles on  master [!] 
❯ curl -fsSL https://get.comtrya.dev | sh
x86_64
Downloading package https://github.com/comtrya/comtrya/releases/download/v0.9.0/comtrya-x86_64-unknown-linux-gnu as /home/zenobius/.dotfiles/comtrya-x86_64-unknown-linux-gnu
Download complete.

============================================================
  The script was run as a user who is unable to write
  to /usr/local/bin. To complete the installation the
  ...

.dotfiles on  master [!?] took 3s 
❯ sudo cp comtrya-x86_64-unknown-linux-gnu /usr/local/bin/comtrya

.dotfiles on  master [!?] took 3s 
❯ comtrya apply -m shells/zsh.yml   
                               
comtrya: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.38' not found (required by comtrya)
comtrya: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.39' not found (required by comtrya)
```

**Example manifest**
Can you provide an example manifest that can demonstrate the problem?

**Steps to reproduce**
Step by step explanation. Please add the output of with `-vvv` flag enabled (be careful not to include any secrets from your config).

**Expected behaviour**
What the expected behaviour in your opinion.

**Environment**
```
❯ lsb_release -a                 
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.5 LTS
Release:        22.04
Codename:       jammy

❯ uname -a                       
Linux 8C5GFY3 5.15.153.1-microsoft-standard-WSL2 #1 SMP Fri Mar 29 23:14:13 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux
```

**Additional information**
Add more context which didn't fit in the fields above.

**Comment by martintc** [comment]
I am gonna investigate some more but initial thoughts are, the linux gnu binaries are compiled on a ubuntu 24.04 github runner. The latest comtrya binaries might be compiled against a newer version of glibc shared object that is available on 24.04. I don't run Ubuntu, but referencing Distrowatch, 22.04 is using 2.35 where as 24.04 is using 2.39.

Looks like this is partially confirmed reading this older issue.
https://github.com/rust-lang/rust-analyzer/issues/13081

**Comment by martintc** [comment]
Also another link for refernce
https://kobzol.github.io/rust/ci/2021/05/07/building-rust-binaries-in-ci-that-work-with-older-glibc.html

**Comment by martintc** [comment]
A few solutions here I am putting for input for others and/or notes to future me.

1. Come up with a new naming scheme and produce binaries for gnu libc linux triplets for each LTS that github offers in runners.

2. Icepuma has linked to some documentation that may allow us to statically compile gnu libc on the linux-gnu triplet.

3. Investigate that the muslc triplet will work fine on gnu-libc systems and just produce binaries for the linux-muslc triplet.

**Comment by airtonix** [comment]
if github runners don't let you run 22.04 (they should), then there is the option to use docker images in a matrix that compiles the binaries.

**Comment by martintc** [comment]
They do still offer and its not an issue of 22.04 isn't available in github runners, I need to make sure I won't be trading off one set of linker problems for another. The binary works on 22.04 now because I changed it back to using the 22.04 runner because it can now link to the appropriate library's version, but is now broken on 24.04 because the library's version is too new.

If that is the case, I'll need a way to spit out binaries for both 22.04 and 24.04 in the release and to detect and choose the right one in the install script that is curl'd to bash.

**Comment by martintc** [comment]
Tested on my arch machine and the musl builds appear to work fine. I changed the script to pull the musl build since that is statically built/linked. This should solve the issue without having to publish and adjust the script for multiple variants of comtrya compiled for multiple LTSs of Ubuntu because of dynamically linking to glibc. 

https://github.com/comtrya/get.comtrya.dev/commit/50b43ed796e636bceb33f1225fb19ce03218a37a

If you wouldn't mind verifying that it works on your machine.
